src/registrar/management/commands/load_organization_data.py

```diff
@@ -0,0 +1,251 @@
+        load = OrganizationDataLoader(args)
+        domain_information_to_update = load.update_organization_data_for_all()
+
+        # Reprompt the user to reinspect before updating DomainInformation
+        proceed = TerminalHelper.prompt_for_execution(
+            system_exit_on_terminate=True,
+            info_to_inspect=f"""
+            ==Master data file==
+            domain_additional_filename: {args.domain_additional_filename}
+
+            ==Organization name information==
+            organization_adhoc_filename: {args.organization_adhoc_filename}
+
+            ==Containing directory==
+            directory: {args.directory}
+
+            ==Proposed Changes==
+            Number of DomainInformation objects to change: {len(domain_information_to_update)}
+            """,
+            prompt_title="Do you wish to load organization data for DomainInformation?",
+        )
+
+        if not proceed:
+            return None
+
+        if len(domain_information_to_update) == 0:
+            logger.error(f"{TerminalColors.MAGENTA}" "No DomainInformation objects exist" f"{TerminalColors.ENDC}")
+            return None
+
+        logger.info(
+            f"{TerminalColors.MAGENTA}"
+            "Preparing to load organization data onto DomainInformation tables..."
+            f"{TerminalColors.ENDC}"
+        )
+        self.update_domain_information(domain_information_to_update, args.debug)
+
+    def update_domain_information(self, desired_objects: List[TransitionDomain], debug):
+        di_to_update = []
+        di_failed_to_update = []
+        di_skipped = []
+
+        # Grab each TransitionDomain we want to change.
+        transition_domains = TransitionDomain.objects.filter(
+            username__in=[item.username for item in desired_objects],
+            domain_name__in=[item.domain_name for item in desired_objects],
+        )
+
+        # This indicates that some form of data corruption happened.
+        if len(desired_objects) != len(transition_domains):
+            raise LoadOrganizationError(code=LoadOrganizationErrorCodes.TRANSITION_DOMAINS_NOT_FOUND)
+
+        # Start with all DomainInformation objects
+        domain_informations = DomainInformation.objects.all()
+        domain_informations_dict = {di.domain.name: di for di in domain_informations}
+
+        # Then, use each domain object to map TransitionDomain <--> DomainInformation
+        # Fetches all DomainInformations in one query.
+        # If any related organization fields have been updated,
+        # we can assume that they modified this information themselves - thus we should not update it.
+        domain_informations = domain_informations.filter(
+            domain__name__in=[td.domain_name for td in transition_domains],
+            address_line1__isnull=True,
+            city__isnull=True,
+            state_territory__isnull=True,
+            zipcode__isnull=True,
+        )
```

Comment:
This is assuming that if the user changed ANY of the given org fields, then we should just skip over it. Basically, it assumes that the later will probably change these fields later if they changed one.

The intent is to minimize accidental data corruption, but it doesn't need to be this way. Instead we can check each field individually and make the determination that way (update if null, leave alone if not). Thoughts on this?